processor.go

```diff
@@ -144,6 +144,16 @@ func (p *processor) calculateRatio(srcWidth, srcHeight int, width, height *int)
 		}
 	}
 
+	if p.aspectRatio == WidthFirst {
+		heightRatio = widthRatio
+		*height = int(math.Round(floatSrcHeight * heightRatio))
+	}
+
+	if p.aspectRatio == HeightFirst {
+		widthRatio = heightRatio
+		*width = int(math.Round(floatSrcWidth * widthRatio))
+	}
+
```

Comment:
The current code looks good overall!
However, it duplicates logic from the `p.aspectRatio == Maintain` block, making it a bit redundant. 
Could you update the `p.aspectRatio == Maintain` block to rewrite `p.aspectRatio` to either `WidthFirst` or `HeightFirst`?

Also, using the `switch` statement here would help make the code cleaner.